sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -265,30 +318,58 @@ export class CommandStem<
   }
 }
 
-export interface SequenceOptions {
-  seqId: string;
-  metadata: Record<string, any>;
-  commands: CommandStem[];
-}
-
-export class Sequence {
-  public readonly seqId: string;
-  public readonly metadata: Record<string, any>;
-  public readonly commands: CommandStem[];
+// @ts-ignore : 'SeqJson' found in JSON Spec
+export class Sequence implements SeqJson {
+  public readonly id: string;
+  // @ts-ignore : 'VariableDeclaration' found in JSON Spec
+  public readonly locals?: [VariableDeclaration, ...VariableDeclaration[]];
+  // @ts-ignore : 'Metadata' found in JSON Spec
+  public readonly metadata: Metadata;
+  // @ts-ignore : 'VariableDeclaration' found in JSON Spec
+  public readonly parameters?: [VariableDeclaration, ...VariableDeclaration[]];
+  // @ts-ignore : 'Step' found in JSON Spec
+  public readonly steps?: Step[];
+  // @ts-ignore : 'Request' found in JSON Spec
+  public readonly requests?: Request[];
+  // @ts-ignore : 'ImmediateCommand' found in JSON Spec
+  public readonly immediate_commands?: ImmediateCommand[];
+  // @ts-ignore : 'HardwareCommand' found in JSON Spec
+  public readonly hardware_commands?: HardwareCommand[];
+  [k: string]: unknown;
+  public readonly commands: CommandStem[]; //  TODO: remove later for Stepp[]
 
   private constructor(opts: SequenceOptions) {
-    this.seqId = opts.seqId;
+    this.id = opts.seqId;
     this.metadata = opts.metadata;
     this.commands = opts.commands;
-  }
 
+    if (opts.locals) {
+      this.locals = opts.locals;
+    }
+    if (opts.parameters) {
+      this.parameters = opts.parameters;
+    }
+    if (opts.steps) {
+      this.steps = opts.steps;
+    }
+    if (opts.requests) {
+      this.requests = opts.requests;
+    }
+    if (opts.immediate_commands) {
+      this.immediate_commands = opts.immediate_commands;
+    }
+    if (opts.hardware_commands) {
+      this.hardware_commands = opts.hardware_commands;
+    }
```

Comment:
How do you guys feel about this or using null coalescing for each value?
```
this.steps = opts.steps ?? undefined;
```

Reply:
I am in favor!!!

Reply:
@cohansen I will make this change when I change CommandStem in the next PR. The `Sequence` Class has more to get `Commands` working for the `Step` field

Reply:
Sounds good!